sample/Screen/LoginScreen.js

```diff
@@ -15,106 +15,205 @@
 
+// Create a config object containing the necessary configurations.
+//for emulator
+const Config = {
+  serverOrigin: 'https://10.0.2.2:9443',
+  signInRedirectURL: 'http://10.0.2.2:8081',
+  clientID: 'iMc7TiIaIFafkd5hA5xf7kGiAWUa',
+  //SignOutURL: "http://10.0.2.2:8081"       (Optional)
+};
 
-class LoginScreen extends React.Component {
+// for device
+//  const Config ={
+//    serverOrigin:"https://192.168.43.29:9443",
+//    signInRedirectURL:"http://192.168.43.29:8081",
+//    clientID: "M1eM7_pf45m0fKJkfCDUTI6Demca",
+//    SignOutURL: "http://192.168.43.29:8081"
+//  };
 
-  handleSubmitPress = async () => {
-    // Create a config object containing the necessary configurations.
-    const Config ={
-      serverOrigin:"https://10.0.2.2:9443",
-      signInRedirectURL:"http://10.0.2.2:8081/Screen/LoginScreen",
-      clientID: "iMc7TiIaIFafkd5hA5xf7kGiAWUa",
-      SignOutURL: "http://10.0.2.2:8081"
+class LoginScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      refreshToken: '',
+      idToken: '',
+      haslogin: false,
+      allowedScopes: '',
+      username: '',
+      sessionState: '',
+      amr: '',
+      at_hash: '',
+      aud: '',
+      azp: '',
+      c_hash: '',
+      exp: '',
+      iat: '',
+      iss: '',
+      nbf: '',
+      sub: '',
     };
+  }
 
-    // initializes the SDK with the config data
-    const _init = await initialize(Config) 
+  componentDidMount() {
+    Linking.addEventListener('url', this.handleAuthUrl);
+  }
+
+  handleAuthUrl = async (Url) => {
+    requestAccessTokenDetails(Url)
+      .then((token) => {
+        // get param of authorization url and return token details
+        this.setState({...token, haslogin: true});
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
 
-     //const _dataLayer = await getDataLayer()
-     // console.log("dataLayer",_dataLayer)
+  handleSubmitPress = async () => {
+    // initializes the SDK with the config data
+    await initialize(Config);
 
     // authenticate with Identity server
-    getAuthorizationURL(Config).then((url) => {
+    getAuthorizationURL(Config)
+      .then((url) => {
+        if (this.state.haslogin === false) {
+          Linking.openURL(url); // Linking the AuthorizeUrl through the internet
+        } else {
+          userInformation().then((user) => {
+            this.setState({...user});
 
-        this.props.navigation.navigate("SignIn",{url:url,config:Config})
-    })
-    .catch((error) => {
+            getDecodedIDToken().then((decodeID) => {
+              this.setState({...decodeID});
+              this.props.navigation.navigate('SignIn', {token: this.state});
+              this.setState({haslogin: false});
+            });
+          });
+        }
+      })
+      .catch((error) => {
         console.error(error);
-    });
+      });
+  };
 
-    }
-  
-    render(){
+  render() {
     return (
       <View style={styles.mainBody}>
```

Comment:
There should be a space between the curly brackets and the characters. Please check and fix other places as well.